webapp/src/auth.ts

```diff
@@ -366,7 +369,7 @@ export function hasIdentity(): boolean {
     // Must read storage for this rather than app theme because this method
     // gets called before experiments are synced to the theme.
     const experimentEnabled = pxt.editor.experiments.isEnabled("identity");
-    return !pxt.BrowserUtils.isPxtElectron() && experimentEnabled && identityProviders().length > 0;
+    return !authDisabled && !pxt.BrowserUtils.isPxtElectron() && experimentEnabled && identityProviders().length > 0;
```

Comment:
Can you query the query string right here?
Might save some plumbing.

otherwise, this seems okay, but I'm a little surprised this solves the issues.
Cloud sync should only be touching projects that make it to the main webapp workspace.
And skillsmap projects live in their own DB so they shouldn't appear there unless the user explicitly said "save to my projects".
And even if auth is disabled while in the skillsmap, it should kick in again as soon as you're in the webapp normally which should sync any pending projects.

So I guess my question is: what behavior are you seeing that this fixes?

Reply:
What I think is happening is:
* skillmap loads the arcade editor in an iframe.
* iframe'd webapp has access to its normal cookies and db, so identity works (it can see the auth cookie and token).
* the editor, with a skillmap project loaded, auto saves, triggering a cloud save.
* cloud-saved projects sync to another client and are written to the regular workspace db.

Reply:
> Can you query the query string right here?

I'd considered that, but this seemed like a better separation of concerns. Let app logic happen at the app level. Auth shouldn't know about skill map.